crates/emitter/scripts/update_opcodes.py

```diff
@@ -100,6 +117,111 @@ def extract_flags(path):
     return flags
 
 
+size_types = {
+    'int8_t': 'i8',
+    'uint8_t': 'u8',
+    'uint16_t': 'u16',
+    'uint24_t': 'u24',
+    'int32_t': 'i32',
+    'uint32_t': 'u32',
+}
+
+
+def extract_types(paths):
+    types = dict()
+
+    def extract_enum(ty):
+        variants_pat = re.compile(
+            r'enum class ' + ty + r' : ([A-Za-z0-9_]+) \{([^}]+)\}', re.M)
+
+        name = '{}.h'.format(ty)
+        with open(paths[name], 'r') as f:
+            content = f.read()
+
+        m = variants_pat.search(content)
+        size_type = m.group(1)
+        if size_type not in size_types:
+            print('{} is not supported'.format(size_type),
+                  file=sys.stderr)
+            sys.exit(1)
+
+        size = size_types[size_type]
+        variants = list(map(lambda s: s.strip(), m.group(2).split(',')))
+
+        types[ty] = {
+            'size': size,
+            'variants': variants
+        }
+
+    def extract_symbols():
+        pat = re.compile(r'MACRO\((.+)\)')
+
+        ty = 'SymbolCode'
+        variants = []
+
+        found = False
+        with open(paths['Symbol.h'], 'r') as f:
+            for line in f:
+                if 'enum class SymbolCode : uint32_t {' in line:
+                    found = True
+
+                m = pat.search(line)
+                if m:
+                    sym = m.group(1)
+                    sym = sym[0].upper() + sym[1:]
+                    variants.append(sym)
```

Comment:
nit: `MACRO` lines are expected only within the `JS_FOR_EACH_WELL_KNOWN_SYMBOL` definition.